doc/guide/pyapi.md

```diff
@@ -0,0 +1,493 @@
+```
+import sling
+
+# Create global store for common definitions.
+commons = sling.Store()
+docschema = sling.DocumentSchema(commons)
+
+# Create global schemas.
+isa = commons["isa"]
+love01 = commons["/pb/love-01"]
+arg0 = commons["/pb/arg0"]
+arg1 = commons["/pb/arg1"]
+person = commons["/saft/person"]
+
+commons.freeze()
+
+# Create record output writer.
+fout = sling.RecordWriter("/tmp/john.rec")
+
+# Add annotated "John loves Mary" example.
+store = sling.Store(commons)
+doc = sling.tokenize("John loves Mary", store=store, schema=docschema)
+john = store.frame({isa: person})
+mary = store.frame({isa: person})
+loves = store.frame({isa: love01, arg0: john, arg1: mary})
+doc.add_mention(0, 1).evoke(john)
+doc.add_mention(1, 2).evoke(loves)
+doc.add_mention(2, 3).evoke(mary)
+doc.update()
+fout.write("0001", doc.frame.data(binary=True))
+
+fin.close()
+fout.close()
+```
+
+The `Document` class has the following methods and properties:
+* `__init__(frame=None, store=None, schema=None)`<br>
+  Creates a new document. If `frame` is None, a new "blank" document is created.
+  Otherwise, the frame is used to initialize the document. If `store` is None,
+  a new store is created for the document. If `schema` is None, a new
+  `DocumentSchema` is created for the store, but it is faster to pass in a
+  pre-initialized document schema when creating new documents.
+* `text` (r/w property)<br>
+  Sets/gets the raw text for the document.
+* `tokens` (r/o property)<br>
+  Returns a list of tokens in the document.
+* `mentions` (r/o property)<br>
+  Returns a list of mentions in the document.
+* `themes` (r/o property)<br>
+  Returns a list of themes in the document.
+* `add_token(text=None, start=None, length=None, brk=SPACE_BREAK)`<br>
+  Adds token to the document.
+* `add_mention(begin, end)`<br>
+  Adds mention to the document.
+* `add_theme(theme)`<br>
+  Adds thematic frame to the document.
+* `update()`<br>
+  Updates the underlying document frame. The `update()` method needs to be
+  called after tokens, mentions, or themes have been added to the document.
+* `phrase(begin, end)`<br>
+  Returns phrase text for a span of tokens.
+* `refresh_annotations()`<br>
+  Re-initializes the document object from the underlying frame.
+
+The `Token` class has the following properties:
+* `index` (r/w int property)<br>
+  Gets/sets the index of the token.
+* `text` (r/w string property)<br>
+  Gets/sets text for token.
+* `start` (r/w int property)<br>
+  Gets/sets the start position in the raw text for the token.
+* `length` (r/w int property)<br>
+  Gets/sets the length (in bytes) of the token in the raw text.
+* `end` (r/o int property)<br>
+  Returns the end position (exclusive) of the token in the raw text.
+* `brk` (r/w int property)<br>
+  Gets/sets the break level for the token, i.e. the spacing between this token
+  and the previous token. The following token break levels are supported:
+    0. `NO_BREAK` no white space between tokens
```

Comment:
This line item appears right after the previous sentence, i.e. not on a separate line by itself.

Reply:
Hmm... seems like my md viewer is a little different than the one on github. I have added an extra newline and change to unordered list.